services/addons/images/count_metric/daily_emailer.py

```diff
@@ -0,0 +1,168 @@
+            if rsu_ip not in rsu_dict:
+                rsu_dict[rsu_ip] = {
+                    "primary_route": "Unknown",
+                    "counts": {},
+                }
+                for t in message_types:
+                    rsu_dict[rsu_ip]["counts"][t] = {"in": 0, "out": 0}
+
+            rsu_dict[rsu_ip]["counts"][type]["in"] = count
+
+
+# Modify the rsu_dict with the specified date range's mongoDB "out" counts for each message type
+# The rsu_dict is modified in place
+def query_mongo_out_counts(rsu_dict, start_dt, end_dt, mongo_db):
+    for type in message_types:
+        collection = mongo_db[f"Ode{type.title()}Json"]
+        # Perform mongoDB aggregate query
+        agg_result = collection.aggregate(
+            [
+                {
+                    "$match": {
+                        "recordGeneratedAt": {
+                            "$gte": start_dt,
+                            "$lt": end_dt,
+                        }
+                    }
+                },
+                {
+                    "$group": {
+                        "_id": f"$metadata.originIp",
+                        "count": {"$sum": 1},
+                    }
+                },
+            ]
+        )
+        for record in agg_result:
+            if not record["_id"]:
+                continue
+            rsu_ip = record["_id"]
+            count = record["count"]
+
+            logging.debug(f"{type.title()} Out count received for {rsu_ip}: {count}")
+
+            # If a RSU that is not in PostgreSQL has counts recorded, add it to the rsu_dict and populate zeroes
+            if rsu_ip not in rsu_dict:
+                rsu_dict[rsu_ip] = {
+                    "primary_route": "Unknown",
+                    "counts": {},
+                }
+                for t in message_types:
+                    rsu_dict[rsu_ip]["counts"][t] = {"in": 0, "out": 0}
+
+            rsu_dict[rsu_ip]["counts"][type]["out"] = count
+
+
+def prepare_rsu_dict():
+    query = (
+        "SELECT to_jsonb(row) "
+        "FROM ("
+        "SELECT ipv4_address, primary_route "
+        "FROM public.rsus "
+        "ORDER BY primary_route ASC, milepost ASC"
+        ") as row"
+    )
+
+    # Query PostgreSQL for the list of SNMP message forwarding configurations tracked in PostgreSQL
+    data = pgquery.query_db(query)
+
+    rsu_dict = {}
+    for row in data:
+        row = dict(row[0])
+        rsu_dict[row["ipv4_address"]] = {
+            "primary_route": row["primary_route"],
+            "counts": {},
+        }
+        for type in message_types:
+            rsu_dict[row["ipv4_address"]]["counts"][type] = {"in": 0, "out": 0}
+    logging.debug(f"Created RSU dictionary: {rsu_dict}")
+
+    return rsu_dict
+
+
+def email_daily_counts(email_body):
+    logging.info("Attempting to send the count emails...")
+    try:
+        email_addresses = os.environ["SMTP_EMAIL_RECIPIENTS"].split(",")

```

Comment:
Not needed for this PR by any means, but what do you think about setting this as configurable from the CV Manager UI? Allow people to subscribe to this tool's email outputs via a new database setting? Then we'd have a simple location for managing/viewing who is getting these and we could add/remove as needed.

Reply:
Yeah that could be pretty neat actually. I will create a backlog ticket for that.